src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -1237,6 +1236,108 @@ private bool IsTransientError(SqlException exc)
             return false;
         }
 
+        private void OpenLoginEnlist(
+            TimeoutTimer timeout,
+            SqlConnectionString connectionOptions,
+            SqlCredential credential,
+            string newPassword,
+            SecureString newSecurePassword,
+            bool redirectedUserInstance)
+        {
+            // Indicates whether we should use primary or secondary first
+            bool useFailoverPartner;
+            string failoverPartner;
+
+            ServerInfo dataSource = new ServerInfo(connectionOptions);
+
+            if (PoolGroupProviderInfo != null)
+            {
+                useFailoverPartner = PoolGroupProviderInfo.UseFailoverPartner;
+                failoverPartner = PoolGroupProviderInfo.FailoverPartner;
+            }
+            else
+            {
+                // Only ChangePassword or SSE User Instance comes through this code path.
+                useFailoverPartner = false;
+                failoverPartner = ConnectionOptions.FailoverPartner;
+            }
+
+            SqlConnectionInternalSourceType sourceType = useFailoverPartner
```

Comment:
This temporary variable was introduced because using the entire ternary inline with the function call was too long.